src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -2103,6 +2103,11 @@ private IEnumerable<TypeInferenceInfo> InferTypeInVariableComponentAssignment(Ex
                         {
                             AddTypeAndName((TupleExpressionSyntax)expr, elementTypesBuilder, elementNamesBuilder);
                         }
+                        else if (expr.IsKind(SyntaxKind.IdentifierName))
+                        {
+                            elementNamesBuilder.Add(((IdentifierNameSyntax)expr).Identifier.ValueText);
+                            elementTypesBuilder.Add(GetTypes(expr).FirstOrDefault().InferredType ?? this.Compilation.ObjectType);
```

Comment:
FirstOrDefault()?.InferredType or First().InferredType

Reply:
```FirstOrDefault()?.InferredType``` is correct here.

FirstOrDefault returns a 'default' struct.  So no need (or ability) to use ```?.```.  ```First()``` can crash since the list may be empty.

Reply:
I see so default is struct, so no need for ?.

Reply:
In htis case yes.